core/Integer.Format.savi

```diff
@@ -42,17 +42,28 @@
       out.push_byte('-')
     )
 
-    // Collect the digits we need to print, from least to most significant.
-    // TODO: Avoid this temporary array allocation here.
-    digits Array(U8) = []
-    while (value > zero) (
-      digits << (value % ten).u8 + '0'
-      value = value / ten
+    // Convert the value to a U64 so we don't have to specialize code by size.
+    // TODO: It would likely be slightly faster to specialize by size here.
+    u64 = value.u64
```

Comment:
this would break for `U128`

Reply:
True, although we don't have `U128` at this time, and there are lots of other places that will also break when/if we add `U128`.

Reply:
We'll catch this by adding tests though I'd prefer when the compiler would complain when precision loss takes place (e.g. converting u64 -> u32).